cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -280,6 +280,70 @@ void generate_depth_remappings(std::map<int, std::pair<std::vector<int>, std::ve
   return total_pages;
 }
 
+/**
+ * @brief Returns a string representation of known encodings
+ *
+ * @param encoding Given encoding
+ * @return STring representation of encoding
+ */
+std::string encoding_to_string(Encoding encoding)
+{
+  switch (encoding) {
+    case Encoding::PLAIN: return "PLAIN";
+    case Encoding::GROUP_VAR_INT: return "GROUP_VAR_INT";
+    case Encoding::PLAIN_DICTIONARY: return "PLAIN_DICTIONARY";
+    case Encoding::RLE: return "RLE";
+    case Encoding::BIT_PACKED: return "BIT_PACKED";
+    case Encoding::DELTA_BINARY_PACKED: return "DELTA_BINARY_PACKED";
+    case Encoding::DELTA_LENGTH_BYTE_ARRAY: return "DELTA_LENGTH_BYTE_ARRAY";
+    case Encoding::DELTA_BYTE_ARRAY: return "DELTA_BYTE_ARRAY";
+    case Encoding::RLE_DICTIONARY: return "RLE_DICTIONARY";
+    case Encoding::BYTE_STREAM_SPLIT: return "BYTE_STREAM_SPLIT";
+    case Encoding::NUM_ENCODINGS: return "NUM_ENCODINGS";
```

Comment:
```suggestion
    case Encoding::NUM_ENCODINGS:
```
You should just let this last case fall through, since `NUM_ENCODINGS` is an enum hack to get at the upper bound, not an actual valid encoding type. :wink: